worked on customers CRUD, updated docs

Psami-wondah opened the pull request.

markessien commented on bigfastapi/customer.py:
```diff
@@ -0,0 +1,125 @@
+
+@app.post("/customers", response_model=customer_schemas.CustomerCreateResponse)
+def create_customer(customer: customer_schemas.CustomerValidator, db: orm.Session = fastapi.Depends(get_db)):
+    organization = db.query(organisation_models.Organization).filter(organisation_models.Organization.name == customer.organization_name).first()
```
You cannot use organisation name. ID, obviously

Psami-wondah replied:
fixed

markessien commented on bigfastapi/customer.py:
```diff
@@ -0,0 +1,125 @@
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
+def get_customers(db: orm.Session = fastapi.Depends(get_db)):
+    customers = db.query(customer_models.Customer).all()
```
What if there are 10 million customers? You return them all?

Psami-wondah replied:
Added pagination

Psami-wondah commented on bigfastapi/schemas/customer_schemas.py:
```diff
@@ -0,0 +1,52 @@
+    date_created: datetime
+    last_updated: datetime
+
+class CustomerUpdate(BaseModel):
```
It is different from the customer. It has optional fields which are None by default. It is for the update endpoint only

Psami-wondah commented on bigfastapi/customer.py:
```diff
@@ -0,0 +1,125 @@
+    else:
+        return {"message": "Organization Doesn't Exist", "customer": {}}
+
+@app.get('/customers', response_model=List[customer_schemas.CustomerInDB])
```
added auth

Psami-wondah commented on bigfastapi/customer.py:
```diff
@@ -0,0 +1,125 @@
+
+@app.get('/customers/organization/{organization_name}', response_model=List[customer_schemas.CustomerInDB])
+def get_customers_by_organization(organization_name: str, db: orm.Session = fastapi.Depends(get_db)):
+    organization = db.query(organisation_models.Organization).filter(organisation_models.Organization.name == organization_name).first()
```
I don't understand

Psami-wondah commented on bigfastapi/customer.py:
```diff
@@ -0,0 +1,125 @@
+    customers = db.query(customer_models.Customer).all()
+    return list(map(customer_schemas.CustomerInDB.from_orm, customers))
+
+@app.get('/customers/organization/{organization_name}', response_model=List[customer_schemas.CustomerInDB])
```
Done

Psami-wondah commented:
I have made some changes as requested
- added pagination for the endpoint which needed it
- named my models properly
- used org id instead of org name
- added auth for the endpoints
